RIOT-OS/RIOT
# cpu/stm32: add ADC support for WB55

### Contribution description

As described in the issue #17260, the ADC in the STM32WB55 has the same basic peripheral as the STM32L4 series, therefore the ADC implementation can be shared between these devices.

Edit: During the creation of this PR I found an error in the P-NUCLEO-WB55 user manual: https://www.st.com/resource/en/user_manual/um2435-bluetooth-low-energy-and-802154-nucleo-pack-based-on-stm32wb-series-microcontrollers-stmicroelectronics.pdf
In Table 10 on page 39, ADC1_IN5 and ADC1_IN6 are switched.

The datasheet is correct: https://www.st.com/resource/en/datasheet/stm32wb55cc.pdf Table 17 page 69.




### Testing procedure



The tests/periph/adc test should run successfully on a P-NUCLEO-WB55 board with this PR applied (and #20756).

To compile the test I ran the following command:
```
BOARD=p-nucleo-wb55 make flash term
```

And this is the console output. The correct ADC lines can be verified by pulling the A0-A5 pins to GND and +3V3 respectively. To test the VBAT channel, SB30 has to be closed (or you can touch SB30 with a jumper cable which is connected to +3V3 on the other side).
```
main(): This is RIOT! (Version: 2024.07-devel-10-g41921a-pr/wb55_adc)

RIOT ADC peripheral driver test

This test will sample all available ADC lines once every 100ms with
a 10-bit resolution and print the sampled results to STDIO


Successfully initialized ADC_LINE(0)
Successfully initialized ADC_LINE(1)
Successfully initialized ADC_LINE(2)
Successfully initialized ADC_LINE(3)
Successfully initialized ADC_LINE(4)
Successfully initialized ADC_LINE(5)
Successfully initialized ADC_LINE(6)
ADC_LINE(0): 292
ADC_LINE(1): 339
ADC_LINE(2): 373
ADC_LINE(3): 396
ADC_LINE(4): 469
ADC_LINE(5): 378
ADC_LINE(6): 179
ADC_LINE(0): 290
ADC_LINE(1): 345
ADC_LINE(2): 368
ADC_LINE(3): 396
ADC_LINE(4): 426
ADC_LINE(5): 369
ADC_LINE(6): 185
ADC_LINE(0): 290
ADC_LINE(1): 347
ADC_LINE(2): 367
ADC_LINE(3): 397
ADC_LINE(4): 412
ADC_LINE(5): 369
ADC_LINE(6): 186
```

### Issues/PRs references

Fixes #17260.
Depends on #20756, because with the mainline RIOT code, the ADC initialization does not work.
This will require a rebase when #20756 is merged because this PR changes the name of adc_l4.c to adc_l4_wb.c
